compiler/rustc_driver/src/lib.rs

```diff
@@ -333,6 +332,8 @@ fn run_compiler(
                 return early_exit();
             }
 
+            queries.global_ctxt()?.enter(|tcx| tcx.output_filenames_raw(()))?;
```

Comment:
Could you add a comment explaining that this is needed for `tcx.output_filenames()` to never fail?

Reply:
Nit: also `let gctxt = queries.global_ctxt()?;` with `gctxt` reused throughout this function would be nicer than calling `queries.global_ctxt()?` multiple times.